### ISSUE TITLE
Unused variable in global scope not found

### ISSUE THREAD

**Issue by sirbrillig** [sirbrillig/phpcs-variable-analysis#37]
A variable defined in a foreach statement and never used is not found.

```php
$foo = [1, 2, 3];
// The next line should report an unused variable
foreach ( $foo as $bar ) {
  echo 'hi';
}
```

**tldr:** The global scope is not scanned because there's no token to indicate when it ends.

**Comment by sirbrillig** [comment]
It looks like this only occurs in the global scope, not within functions. I think there might be an issue with which scope these variables are put into.

**Comment by sirbrillig** [comment]
💡 I think this is because the global scope is not being checked for unused variables. And the reason for that is that there is no token to identify the end of a file, so it's hard to determine when to look for unused variables in that scope.

**Comment by stuwest** [comment]
> I think this is because the global scope is not being checked for unused variables

I just noticed this last night was going to post a new issue with this example:

```php
<?php

$missed_this_unused_variable = 'uh-oh';

function placeholder() {
    $got_this_unused_variable = 'sweet';
}
```

**Comment by stuwest** [comment]
not sure if fixing the global scope issue would naturally handle this, but would also be great to identify unused imports of functions, constants, via `use`.
